### PR TITLE
Remove eagle related code and reference since eagle is retired.

### LINKED ISSUE #470: Remove Eagle specific code
Now that Eagle is gone we should probably remove the Eagle specific stuff:

- create_eagle_env.sh
- `EagleBatch` class from hpc.py
- `eagle_cli` function from hpc.py
- `buildstock_eagle` entry point from setup.py
- documentation about Eagle

I think that's all, but I may have missed something

### PR DESCRIPTION
**Pull request by rajeee**
Fixes #470.

## Pull Request Description

Remove all eagle related code and references since Eagle is already retired from NREL HPC.

Migration guide:
The following changes are necessary on the yaml file after this PR is merged:
Change schema version to 0.5.
Change eagle block to kestrel block. The specification is the same, so no changes are needed for the arguments in the block.

## Checklist

Not all may apply

- [x] Code changes (must work)
- [x] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [x] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/coverage.yml` as necessary.
- [x] All other unit and integration tests passing
- [x] Update validation for project config yaml file changes
- [x] Update existing documentation
- [x] Run a small batch run on Kestrel/Eagle to make sure it all works if you made changes that will affect Kestrel/Eagle
- [x] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: buildstockbatch/schemas/v0.4.yaml [hunk 1]
```diff
@@ -5,8 +5,8 @@ weather_files_path: str(required=False)
 weather_files_url: str(required=False)
 sampler: include('sampler-spec', required=True)
 workflow_generator: include('workflow-generator-spec', required=True)
-eagle: include('hpc-spec', required=False)
 kestrel: include('hpc-spec', required=False)
+eagle: include('hpc-spec', required=False)
```
Comment (nmerket) [inline buildstockbatch/schemas/v0.4.yaml:9]: It just seems like you moved this. Shouldn't it be removed?

File: buildstockbatch/schemas/v0.5.yaml
Comment (nmerket) [inline buildstockbatch/schemas/v0.5.yaml:1]: Ah okay, a new schema without `eagle`.

File: docs/changelog/changelog_0_18.rst [hunk 2]
```diff
@@ -60,9 +60,9 @@
 
         Added option to activate custom gems in the singularity container in eagle-based simulations by setting the
         ``custom_gem`` key in the ``baseline`` section of the configuration yaml to ``True``. This is implemented via
-        the ``bundle`` and ``bundle_path`` options in the OpenStudio CLI. See :func:`~.EagleBatch.run_building` for
-        additional implementation details.
-
+        the ``bundle`` and ``bundle_path`` options in the OpenStudio CLI.
+        See `run_building <https://github.com/NREL/buildstockbatch/blob/61c00cc4139769939f16db9b62f3a7dc58f1f46e/buildstockbatch/eagle.py#L334>`_
+        for additional implementation details.
```
Comment (nmerket) [inline docs/changelog/changelog_0_18.rst:65]: Makes sense to pin it to a specific commit.

**Review by nmerket — APPROVED** [review]
Suggestion: add something to a migration guide about removing the `eagle` block and migrating to Kestrel for those with project files that haven't been run since the Eagle days.

**Comment by rajeee** [comment]
> Suggestion: add something to a migration guide about removing the `eagle` block and migrating to Kestrel for those with project files that haven't been run since the Eagle days.

Yes, I will add that to the migration guide. I will add that text to the PR description.

**Pull request merged**